plugins/inputs/nats_consumer/nats_consumer_test.go

```diff
@@ -0,0 +1,638 @@
+package nats_consumer
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/require"
+)
+
+type FakeClient struct {
+	url             string
+	closed          bool
+	queueSubscribeF func(sub, group string, callback nats.MsgHandler) (*nats.Subscription, error)
+
+	closeCalled          int
+	queueSubscribeCalled int
+	jetStreamCalled      int
+}
+
+func (f *FakeClient) QueueSubscribe(subj string, group string, callback nats.MsgHandler) (*nats.Subscription, error) {
+	f.queueSubscribeCalled++
+	return f.queueSubscribeF(subj, group, callback)
+}
+
+func (f *FakeClient) JetStream(...nats.JSOpt) (nats.JetStreamContext, error) {
+	f.jetStreamCalled++
+	return &FakeJetStreamContext{}, nil
+}
+
+func (f *FakeClient) ConnectedUrl() string {
```

Comment:
Method Name cannot be changed to ConnectedURL(). The original type is defined in the nats lib itself and the interface is only there to make the consumer testable. 

https://github.com/nats-io/nats.go/blob/main/nats.go#L1965